Catch expected non-200 errors on submit.

kshakir opened the pull request.

kshakir commented:
Code tested here: https://travis-ci.org/broadinstitute/cromwell/builds/259495692

kshakir commented:
Tests passed here: https://travis-ci.org/broadinstitute/cromwell/builds/262089543

cjllanwarne commented on src/main/resources/standardTestCases/invalid_options_json.test:
```diff
@@ -0,0 +1,15 @@
+  statusCode: 400
+  message: """{
+  "status": "fail",
+  "message": "Error(s): Invalid workflow options provided: Unexpected end-of-input at input index 72 (line 3, position 1), expected '}':\n\n^\n"
```
ToL: I guess that console layout magic doesn't age so well when it's used in a JSON response!

cjllanwarne commented on src/main/resources/standardTestCases/invalid_options_json/invalid_options_json.wdl:
```diff
@@ -0,0 +1,9 @@
+task put_that_coffee_down {
+  command {
+    echo "coffee's for valid options json only"
```
Yikes! Do you need a coffee break!?

cjllanwarne commented on src/main/scala/centaur/test/Test.scala:
```diff
@@ -247,6 +276,29 @@ object Operations {
+            ()
+          } else {
+            throw new RuntimeException(
+              s"""|Expected
```
Since you're multi-lining anyway, might as well put the `Expected` on a new line so it's easier to see amongst all the stack tracings

cjllanwarne commented on src/main/scala/centaur/test/submit/SubmitResponse.scala:
```diff
@@ -0,0 +1,39 @@
+    }
+  }
+
+  private def toErrorOr[A](result: Result[A]): ErrorOr[A] = {
```
Is this really a one-off for `SubmitResponse` that's never used anywhere else?

cjllanwarne reviewed (approved):
Pardon my ToLs, this is thumbs worthy: 👍

Horneth commented:
👍 

[![Approved with PullApprove](https://img.shields.io/badge/pullapprove-approved-brightgreen.svg)](https://pullapprove.com/broadinstitute/centaur/pull-request/226/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=broadinstitute/centaur)

kshakir committed:
Catch expected non-200 errors on submit.
Using new cromwell client that allow invalid json to pass through.
Bumped other dependencies to latest versions.

kshakir commented:
Merging as pullapprove isn't configured correctly yet.
